### ISSUE TITLE
Lograge failing with Rails 7.0

### ISSUE THREAD

**Issue by leifg** [iMacTia/lograge-sql#56]
With the latest update to 2.3.1 it seems like backwards compatibility broke. I'm using Rails 7.0.8 and I'm getting these log entries whenever I'm executing a DB query:

```
[2023-11-18T00:37:38.143+00:00] ERROR: Could not log "sql.active_record" event. NoMethodError: undefined method `+' for nil:NilClass ["/gems/ruby/3.2.0/gems/lograge-sql-2.3.1/lib/lograge/active_record_log_subscriber.rb:24:
in `increase_runtime_duration'", "/gems/ruby/3.2.0/gems/lograge-sql-2.3.1/lib/lograge/active_record_log_subscriber.rb:9:in `sql'", "/gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/subscriber.rb:149:in `finish'
", "/gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/log_subscriber.rb:115:in `finish'", "/gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/notifications/fanout.rb:211:in `finish'", "/gems/ruby/3.2.0
/gems/activesupport-7.0.8/lib/active_support/notifications/fanout.rb:76:in `block in finish'", "/gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/notifications/fanout.rb:91:in `block in iterate_guarding_exceptio
ns'", "/gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/notifications/fanout.rb:90:in `each'", "/gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/notifications/fanout.rb:90:in `iterate_guarding_excep
tions'", "/gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/notifications/fanout.rb:76:in `finish'", "/gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/notifications/instrumenter.rb:49:in `finish_with
_state'", "/gems/ruby/3.2.0/gems/activesupport-7.0.8/lib/active_support/notifications/instrumenter.rb:30:in `instrument'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract_adapter.rb
:743:in `log'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/postgresql/database_statements.rb:46:in `execute'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_ada
pters/postgresql_adapter.rb:296:in `initialize'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/postgresql_adapter.rb:36:in `new'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_rec
ord/connection_adapters/postgresql_adapter.rb:36:in `postgresql_connection'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:656:in `public_send'", "/gems/ruby/3
.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:656:in `new_connection'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_p
ool.rb:700:in `checkout_new_connection'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:679:in `try_to_checkout_new_connection'", "/gems/ruby/3.2.0/gems/activer
ecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:640:in `acquire_connection'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:341:in
 `checkout'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapters/abstract/connection_pool.rb:181:in `connection'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_adapter
s/abstract/connection_handler.rb:211:in `retrieve_connection'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/connection_handling.rb:313:in `retrieve_connection'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/li
b/active_record/connection_handling.rb:280:in `connection'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/model_schema.rb:407:in `table_exists?'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/
attribute_methods/primary_key.rb:95:in `get_primary_key'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/attribute_methods/primary_key.rb:83:in `reset_primary_key'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/
lib/active_record/attribute_methods/primary_key.rb:71:in `primary_key'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/relation/delegation.rb:93:in `primary_key'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/li
b/active_record/relation/finder_methods.rb:575:in `ordered_relation'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/active_record/relation/finder_methods.rb:174:in `last'", "/gems/ruby/3.2.0/gems/activerecord-7.0.8/lib/ac
tive_record/querying.rb:22:in `last'", "/gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/commands/runner/runner_command.rb:46:in `<main>'", "/gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/commands/runner/runner_command.rb:46
:in `eval'", "/gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/commands/runner/runner_command.rb:46:in `perform'", "/gems/ruby/3.2.0/gems/thor-1.3.0/lib/thor/command.rb:28:in `run'", "/gems/ruby/3.2.0/gems/thor-1.3.0/lib/tho
r/invocation.rb:127:in `invoke_command'", "/gems/ruby/3.2.0/gems/thor-1.3.0/lib/thor.rb:527:in `dispatch'", "/gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/command/base.rb:87:in `perform'", "/gems/ruby/3.2.0/gems/railties-
7.0.8/lib/rails/command.rb:48:in `invoke'", "/gems/ruby/3.2.0/gems/railties-7.0.8/lib/rails/commands.rb:18:in `<main>'", "<internal:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'", "<interna
l:/usr/local/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'", "/gems/ruby/3.2.0/gems/bootsnap-1.17.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'", "bin/rails:4:in `<main>'"]
[2023-11-18T00:37:38.145+00:00]
```

It seems like [this commit](https://github.com/iMacTia/lograge-sql/commit/59e842fbe2a5310666f0874b819392f012ed0471) is responsible.

**Comment by iMacTia** [comment]
Thank you for reporting this @leifg, I'm sorry about that!
This seems to be an issue with initialisation. Although `` has been available since Rails 7.2, it seems like it wasn't correctly initialised until [this fix](https://github.com/rails/rails/commit/f56b4187be56304857b57ec7159a8e2dc2b91dea#diff-fc921d14f4dc8556c4b8cd785f350ec728fe07c8b884fbc3a0b4cbc9ec5eafffR13) was released in Rails 7.1, so using an older version of Rails will incur in this issue.

This is annoying but I guess the only thing we can do to support both Rails 7.2+ and older is do the initialisation ourselves 😢 . I'll fix this now

**Comment by iMacTia** [comment]
@leifg this is on it's way to Rubygems as release 2.3.2, I'd appreciate if you could give it a try when you get a chance and let me know if it fixes the issue for you 🙏

**Comment by leifg** [comment]
Issue is fixed in 2.3.2

Thank you for jumping on this so quickly!
